core/src/main/java/org/apache/calcite/rex/RexPermuteInputsShuttle.java

```diff
@@ -54,14 +55,41 @@ public class RexPermuteInputsShuttle extends RexShuttle {
   public RexPermuteInputsShuttle(
       Mappings.TargetMapping mapping,
       RelNode... inputs) {
-    this(mapping, fields(inputs));
+    this(mapping, fields(inputs), false);
+  }
+
+  /**
+   * Creates a RexPermuteInputsShuttle with target type matching control.
+   *
+   * <p>The mapping provides at most one target for every source. If a source
```

Comment:
I think this paragraph belongs in the documentation for the mapping parameter.
I don't think you should discuss here what TargetMapping does, that should belong to the documentation of TargetMapping itself. What you should say is what this function assumes about the mapping. Perhaps you mean to say that every RexInputRef in the expression that will be analyzed by this shuttle needs to be a key in this mapping?

Reply:
what I suggested here is that the second phrase and third should be removed. (If a source... unique target). They do not document what this method does, they document what getTarget does.

Reply:
Done!